component/integration/aws/s3/deploy.go

```diff
@@ -94,6 +96,35 @@ func Deploy(ctx mongo.SessionContext, actionID primitive.ObjectID, source app.In
 		return err
 	}
 
+	if err = invalidateCache(target, session); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func invalidateCache(target app.Instance, session *session.Session) error {
+	if target.CloudFrontId == "" {
```

Comment:
I would think this check would be outside this function since this function implies that it will validate cache, not that it will if it can. Also, would using the `len` function be more efficient than string comparison?

Reply:
making that change